### PR TITLE
Trying step0 in javascript, readline not working

### PR DESCRIPTION
**Pull request by DjebbZ**
Hello @kanaka,

I'm trying to make a lisp in Javascript. I'm failing at step0, and would love your help to understand why my implementation doesn't work. Basically it doesn't pass the tests, they all fail after the timeout. The test output is below.

```
-> make test^javascript^step0
----------------------------------------------
Testing test^javascript^step0, step file: javascript/step0_repl.js, test file: tests/step0_repl.mal
Running: ../runtest.py  ../tests/step0_repl.mal -- node ../javascript/step0_repl.js 
Started with:

TEST: hello world -> ['','hello world'] -> FAIL (line 2):
    Expected : ['hello world\r\nhello world', 'hello world\r\nhello world\r\nhello world']
    Got      : None
TEST: abcABC123 -> ['','abcABC123'] -> FAIL (line 5):
    Expected : ['abcABC123\r\nabcABC123', 'abcABC123\r\nabcABC123\r\nabcABC123']
    Got      : None
TEST: ;:() []{}"'* -> ['',';:() []{}"\'*'] -> FAIL (line 8):
    Expected : [';:() []{}"\'*\r\n;:() []{}"\'*', ';:() []{}"\'*\r\n;:() []{}"\'*\r\n;:() []{}"\'*']
    Got      : None
TEST: hello world abcdefghijklmnopqrstuvwxyz ABCDEFGHIJKLMNOPQRSTUVWXYZ 0123456789 (;:() []{}"'* ;:() []{}"'* ;:() []{}"'*) -> ['','hello world abcdefghijklmnopqrstuvwxyz ABCDEFGHIJKLMNOPQRSTUVWXYZ 0123456789 (;:() []{}"\'* ;:() []{}"\'* ;:() []{}"\'*)'] -> FAIL (line 13):
    Expected : ['hello world abcdefghijklmnopqrstuvwxyz ABCDEFGHIJKLMNOPQRSTUVWXYZ 0123456789 (;:() []{}"\'* ;:() []{}"\'* ;:() []{}"\'*)\r\nhello world abcdefghijklmnopqrstuvwxyz ABCDEFGHIJKLMNOPQRSTUVWXYZ 0123456789 (;:() []{}"\'* ;:() []{}"\'* ;:() []{}"\'*)', 'hello world abcdefghijklmnopqrstuvwxyz ABCDEFGHIJKLMNOPQRSTUVWXYZ 0123456789 (;:() []{}"\'* ;:() []{}"\'* ;:() []{}"\'*)\r\nhello world abcdefghijklmnopqrstuvwxyz ABCDEFGHIJKLMNOPQRSTUVWXYZ 0123456789 (;:() []{}"\'* ;:() []{}"\'* ;:() []{}"\'*)\r\nhello world abcdefghijklmnopqrstuvwxyz ABCDEFGHIJKLMNOPQRSTUVWXYZ 0123456789 (;:() []{}"\'* ;:() []{}"\'* ;:() []{}"\'*)']
    Got      : None
FAILURES: 4
make: *** [test^javascript^step0] Error 2
```

After many failing I decided to "cheat" and read your implementation. This is where I noticed that basically 2 things that differed with mine :
- you wrap a binary tool with an ffi to do the "readline" stuff, adding history along the way ; I just use the core node.js module `readline`.
- you expose a printer that basically wraps `console.log` ; I just use `console.log` directly.

I think the problem I have is with "readline", but I have no clue why it's not working. I also don't understand why you didn't use the core readline module.

I wanted to read the test protocol but I'm not proficient (at all) with Python to understand and make sure my code pass the tests.

So basically, I'm requesting a code review :) That's why I made a pull request, just so you could read the code easily.

### CODE REVIEW

**Comment by DjebbZ** [comment]
Note that the original javascript tests pass on my machine (OSX 10.8)

**Comment by kanaka** [comment]
Resolved via discussion on #mal Answer turns out to be adding terminal:false when initializing node readline:

```
var readline = require("readline"),
    rl = readline.createInterface({
        terminal: false,
        input: process.stdin,
        output: process.stdout
    });
```

**Comment by DjebbZ** [comment]
Also needed to print using `process.stdout` instead of `console`, like so :

```
rl.on("line", function(line) {
    process.stdout.write(rep(line));
    rl.prompt();
})
```

Thanks @kanaka for your kind help.
